File: python/algorithms/easy/compareTheTriplets.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Compare the Triplets - HackerRank
Test description: https://www.hackerrank.com/challenges/compare-the-triplets/problem
"""
#---------- functions ----------
def compareTriplets(a: list[int], b: list[int]) -> list[int]:
    """
    Compare two triplets and return the scores.

    Parameters:
    a (list[int]): First triplet.
    b (list[int]): Second triplet.

    Returns:
    list[int]: Scores of both triplets.
    """
    result: list[int] = [0, 0]

    for idx, el in enumerate(a):
        if el > b[idx]:
            result[0] += 1
        elif el < b[idx]:
            result[1] += 1

    return result

File: go/algorithms/easy/compareTheTriplets.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-13
Last modified: 2026-06-13
Description: Compare the Triplets - HackerRank
See: https://www.hackerrank.com/challenges/compare-the-triplets/problem
*/
//--------- package ---------
package easy


//--------- functions ---------
// compareTriplets compares the elements of two arrays and returns an array with the scores of each array.
func compareTriplets(a []int32, b []int32) []int32 {
	var res []int32 = []int32{0, 0};

	for i, value := range a {
		if value > b[i] {
			res[0]++;
		} else if value < b[i] {
			res[1]++;
		}//end if
	}//end for

	return res;
}//end compareTriplets()